src/View/ColumnView.vala

```diff
@@ -120,40 +120,45 @@ namespace FM {
         }
 
         protected override bool handle_primary_button_click (Gdk.EventButton event, Gtk.TreePath? path) {
-            unowned GOF.File file = selected_files.data;
-            bool is_folder = file.is_folder ();
+            bool result = true;
 
-            selected_folder = null;
+            if (selected_files != null) {
+                unowned GOF.File file = selected_files.data;
+                bool is_folder = file.is_folder ();
 
-            if (!is_folder || !Preferences.settings.get_boolean ("single-click")) {
-                return base.handle_primary_button_click (event, path);
-            }
+                selected_folder = null;
 
-            selected_folder = file;
-            bool result = true;
-
```

Comment:
This is (I think) a simpler solution:
```
 protected override bool handle_primary_button_click (Gdk.EventButton event, Gtk.TreePath? path) {
            unowned GOF.File? file = null;
            if (selected_files != null) {
                file = selected_files.data;
            }

            selected_folder = null;

            if (file == null || !file.is_folder () || !Preferences.settings.get_boolean ("single-click")) {
                return base.handle_primary_button_click (event, path);
            }

            selected_folder = file;
            bool result = true;

```

the rest of the function being unaltered.  The base function just returns true so a null selection there does not matter.